File: app/src/main/java/com/example/ui/screens/PosScreen.kt
package com.example.ui.screens

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Fastfood
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.data.model.CategoryEntity
import com.example.data.model.MenuEntity
import com.example.data.model.TableEntity
import com.example.ui.components.CartBottomSummary
import com.example.ui.components.CartDetailSheet
import com.example.ui.components.CategoryFilterTabs
import com.example.ui.components.MenuCard
import com.example.ui.components.OrderItemNotesDialog
import com.example.ui.components.TopAppBarHeader
import com.example.ui.viewmodel.PosUiState
import com.example.ui.viewmodel.RestaurantViewModel

@Composable
fun PosScreen(
    uiState: PosUiState,
    categories: List<CategoryEntity>,
    menus: List<MenuEntity>,
    tables: List<TableEntity>,
    viewModel: RestaurantViewModel,
    onNavigateToTables: () -> Unit,
    modifier: Modifier = Modifier
) {
    var showCartDetail by remember { mutableStateOf(false) }

    Box(modifier = modifier.fillMaxSize()) {
        Column(modifier = Modifier.fillMaxSize()) {
            // Header with search & table selector
            TopAppBarHeader(
                searchQuery = uiState.searchQuery,
                onSearchChange = { viewModel.setSearchQuery(it) },
                selectedTable = uiState.selectedTable,
                onTableClick = onNavigateToTables
            )

            // Category Filter Chips
            CategoryFilterTabs(
                categories = categories,
                selectedCategoryId = uiState.selectedCategoryId,
                onCategorySelected = { viewModel.selectCategory(it) }
            )

            // Menu Grid
            if (menus.isEmpty()) {
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .weight(1f)
                        .padding(24.dp),
                    contentAlignment = Alignment.Center
                ) {
                    Column(horizontalAlignment = Alignment.CenterHorizontally) {
                        Icon(
                            imageVector = Icons.Default.Fastfood,
                            contentDescription = null,
                            tint = MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.5f),
                            modifier = Modifier.size(54.dp)
                        )
                        Spacer(modifier = Modifier.height(12.dp))
                        Text(
                            text = "Menu tidak ditemukan",
                            fontSize = 16.sp,
                            fontWeight = FontWeight.Bold,
                            color = MaterialTheme.colorScheme.onSurface
                        )
                        Spacer(modifier = Modifier.height(4.dp))
                        Text(
                            text = "Coba ubah kata kunci pencarian atau pilih kategori lain.",
                            fontSize = 13.sp,
                            color = MaterialTheme.colorScheme.onSurfaceVariant,
                            textAlign = TextAlign.Center
                        )
                    }
                }
            } else {
                LazyVerticalGrid(
                    columns = GridCells.Adaptive(minSize = 160.dp),
                    contentPadding = PaddingValues(
                        start = 16.dp,
                        end = 16.dp,
                        top = 8.dp,
                        bottom = if (uiState.cartItems.isNotEmpty()) 90.dp else 16.dp
                    ),
                    horizontalArrangement = Arrangement.spacedBy(12.dp),
                    verticalArrangement = Arrangement.spacedBy(12.dp),
                    modifier = Modifier
                        .fillMaxWidth()
                        .weight(1f)
                        .testTag("menu_grid")
                ) {
                    items(menus, key = { it.id }) { menu ->
                        val cartItem = uiState.cartItems.find { it.menu.id == menu.id }
                        MenuCard(
                            menu = menu,
                            cartItem = cartItem,
                            onAddToCart = { viewModel.addToCart(menu) },
                            onQuantityChange = { qty -> viewModel.updateCartQuantity(menu.id, qty) },
                            onOpenNotes = { item -> viewModel.openNotesDialog(item) }
                        )
                    }
                }
            }
        }

        // Floating Sticky Cart Summary
        CartBottomSummary(
            uiState = uiState,
            onOpenCartDetail = { showCartDetail = true },
            modifier = Modifier.align(Alignment.BottomCenter)
        )
    }

    // Cart Details Sheet
    if (showCartDetail && uiState.cartItems.isNotEmpty()) {
        CartDetailSheet(
            uiState = uiState,
            tables = tables,
            onDismiss = { showCartDetail = false },
            onQuantityChange = { menuId, qty -> viewModel.updateCartQuantity(menuId, qty) },
            onRemoveItem = { menuId -> viewModel.removeFromCart(menuId) },
            onClearCart = {
                viewModel.clearCart()
                showCartDetail = false
            },
            onOpenNotes = { item -> viewModel.openNotesDialog(item) },
            onCustomerNameChange = { viewModel.setCustomerName(it) },
            onOrderTypeChange = { viewModel.setOrderType(it) },
            onTableSelect = { viewModel.selectTable(it) },
            onDiscountChange = { viewModel.setDiscountPercent(it) },
            onProceedPayment = {
                showCartDetail = false
                viewModel.openPaymentDialog()
            },
            onSavePendingOrder = {
                showCartDetail = false
                viewModel.processPayment(
                    paymentMethod = com.example.data.model.PaymentMethod.CASH,
                    amountPaid = 0.0,
                    isPaidNow = false
                )
            }
        )
    }

    // Notes Dialog
    uiState.activeNotesItem?.let { notesItem ->
        OrderItemNotesDialog(
            item = notesItem,
            onSaveNotes = { note -> viewModel.setItemNotes(notesItem.menu.id, note) },
            onDismiss = { viewModel.closeNotesDialog() }
        )
    }
}

File: lib/screens/pos_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../models/cart_item_model.dart';
import '../models/order_model.dart';
import '../providers/restaurant_provider.dart';
import '../theme/app_colors.dart';
import '../widgets/cart_bottom_summary.dart';
import '../widgets/cart_detail_sheet.dart';
import '../widgets/category_filter_tabs.dart';
import '../widgets/menu_card.dart';
import '../widgets/order_item_notes_dialog.dart';
import '../widgets/top_app_bar_header.dart';

class PosScreen extends StatelessWidget {
  const PosScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final provider = context.watch<RestaurantProvider>();
    final menus = provider.filteredMenus;

    return Stack(
      children: [
        Column(
          children: [
            // Top App Bar Header with Search & Table Selector
            TopAppBarHeader(
              searchQuery: provider.searchQuery,
              onSearchChange: (query) => provider.setSearchQuery(query),
              selectedTable: provider.selectedTable,
              onTableClick: () => provider.setTab(AppNavTab.tables),
            ),

            // Category Filter Chips
            CategoryFilterTabs(
              categories: provider.categories,
              selectedCategoryId: provider.selectedCategoryId,
              onCategorySelected: (catId) => provider.selectCategory(catId),
            ),

            // Menu Grid
            Expanded(
              child: menus.isEmpty
                  ? Center(
                      child: Padding(
                        padding: const EdgeInsets.all(24),
                        child: Column(
                          mainAxisAlignment: MainAxisAlignment.center,
                          children: [
                            Icon(
                              Icons.fastfood,
                              size: 54,
                              color: AppColors.textSecondaryLight.withValues(alpha: 0.5),
                            ),
                            const SizedBox(height: 12),
                            const Text(
                              'Menu tidak ditemukan',
                              style: TextStyle(
                                fontSize: 16,
                                fontWeight: FontWeight.bold,
                                color: AppColors.textPrimaryLight,
                              ),
                            ),
                            const SizedBox(height: 4),
                            const Text(
                              'Coba ubah kata kunci pencarian atau pilih kategori lain.',
                              style: TextStyle(
                                fontSize: 12,
                                color: AppColors.textSecondaryLight,
                              ),
                              textAlign: TextAlign.center,
                            ),
                          ],
                        ),
                      ),
                    )
                  : GridView.builder(
                      padding: EdgeInsets.fromLTRB(
                        16,
                        8,
                        16,
                        provider.cartItems.isNotEmpty ? 90 : 16,
                      ),
                      gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
                        crossAxisCount: 2,
                        crossAxisSpacing: 12,
                        mainAxisSpacing: 12,
                        childAspectRatio: 0.76,
                      ),
                      itemCount: menus.length,
                      itemBuilder: (context, index) {
                        final menu = menus[index];
                        final cartItem = provider.cartItems.cast<CartItemModel?>().firstWhere(
                              (it) => it?.menu.id == menu.id,
                              orElse: () => null,
                            );

                        return MenuCard(
                          menu: menu,
                          cartItem: cartItem,
                          onAddToCart: () => provider.addToCart(menu),
                          onQuantityChange: (qty) =>
                              provider.updateCartQuantity(menu.id!, qty),
                          onOpenNotes: (item) {
                            showDialog(
                              context: context,
                              builder: (ctx) => OrderItemNotesDialog(
                                item: item,
                                onSaveNotes: (notes) =>
                                    provider.setItemNotes(item.menu.id!, notes),
                                onDismiss: () => Navigator.pop(ctx),
                              ),
                            );
                          },
                        );
                      },
                    ),
            ),
          ],
        ),

        // Sticky Bottom Summary Bar
        Align(
          alignment: Alignment.bottomCenter,
          child: CartBottomSummary(
            totalItemCount: provider.totalItemCount,
            grandTotal: provider.grandTotal,
            orderLabel: provider.selectedTable != null
                ? 'Meja ${provider.selectedTable!.tableNumber}'
                : provider.orderType.displayName,
            onOpenCartDetail: () {
              showModalBottomSheet(
                context: context,
                isScrollControlled: true,
                backgroundColor: Colors.transparent,
                builder: (ctx) => CartDetailSheet(
                  provider: provider,
                  onProceedPayment: () {
                    Navigator.pop(ctx);
                    provider.openPaymentDialog();
                  },
                  onSavePendingOrder: () {
                    Navigator.pop(ctx);
                    provider.processPayment(
                      paymentMethod: PaymentMethod.cash,
                      amountPaid: 0.0,
                      isPaidNow: false,
                    );
                  },
                ),
              );
            },
          ),
        ),
      ],
    );
  }
}
